rfcs/0017-intensional-store.md

```diff
@@ -0,0 +1,110 @@
+---
+feature: intensional_store
+start-date: 2017-08-11
+author: Wout.Mertens@gmail.com
+co-authors: (find a buddy later to help our with the RFC)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+One paragraph explanation of the feature.
+
+# Motivation
+[motivation]: #motivation
```

Comment:
I would really like to see a lot more detail. This is a huge, invasive change and it would be good to have some assurance that it will actually be worthwhile

Reply:
I hope the new section provides this :)